src/com/intellij/rt/coverage/data/ClassData.java

```diff
@@ -34,6 +34,7 @@
 
   /** Storage for line and branch hits in new tracing mode. */
   private volatile int[] myHitsMask;
+  private volatile boolean[] myLinesTrace;
```

Comment:
please leave a comment what is it or give a better name; also can't we use only this array (of ints then) and on save append to the common hitsMask? Or won't it save some time?